src/qutip_qip/compiler/gatecompiler.py

```diff
@@ -247,6 +249,18 @@ def _concatenate_pulses(
                 compiled_tlist[pulse_ind].append(execution_time)
                 compiled_coeffs[pulse_ind].append(coeffs)
 
+        final_time = np.max([tlist[-1] for tlist in compiled_tlist])
```

Comment:
It looks like this is unrelated to QuTiP 5. I'm happy for it to stay in this PR potentially, but what does it add / do?

Reply:
This is actually related to the fact that evaluation out of the tlist will give non-zero values in qutip 5. Because of this, I need to determine the final time and add some zeros to the end.